client/asset/eth/contractor.go

```diff
@@ -171,47 +187,47 @@ func (c *contractorV0) redeem(txOpts *bind.TransactOpts, redemptions []*asset.Re
 }
 
 // swap retrieves the swap info from the read-only swap method.
-func (c *contractorV0) swap(ctx context.Context, secretHash [32]byte) (*dexeth.SwapState, error) {
+func (c *contractorV0) status(ctx context.Context, contract *dex.SwapContractDetails) (step dexeth.SwapStep, secret [32]byte, blockNumber uint32, err error) {
```

Comment:
A preferable self-documenting design IMO would be to leave as many `contractorV0` methods unchanged as possible except for their name (e.g. `status` -> `statusImpl`) and have them wrapped with the more confusing function signature with the `SwapContractDetails` argument.

Reply:
mmm. OK.